cmd/snap/cmd_info.go

```
@@ -0,0 +1,191 @@
+}
+
+var shortInfoHelp = i18n.G("show detailed information about a snap")
+var longInfoHelp = i18n.G(`
+The info command shows detailed information about a snap, be it by name or by path.`)
+
+func init() {
+	addCommand("info",
+		shortInfoHelp,
+		longInfoHelp,
+		func() flags.Commander {
+			return &infoCmd{}
+		}, map[string]string{
+			"verbose": i18n.G("Include a verbose list of a snap's notes (otherwise, summarise notes)"),
+		}, nil)
+}
+
+func norm(path string) string {
+	path = filepath.Clean(path)
+	if osutil.IsDirectory(path) {
+		path = path + "/"
+	}
+
+	return path
+}
+
+func tryDirect(w io.Writer, path string, verbose bool) bool {
+	path = norm(path)
+
+	snapf, err := snap.Open(path)
+	if err != nil {
+		return false
+	}
+
+	info, err := snap.ReadInfoFromSnapFile(snapf, nil)
+	if err != nil {
+		return false
+	}
+	fmt.Fprintf(w, "path:\t%q\n", path)
+	fmt.Fprintf(w, "name:\t%s\n", info.Name())
+	fmt.Fprintf(w, "summary:\t%q\n", info.Summary())
+
+	var notes *Notes
+	if verbose {
+		fmt.Fprintln(w, "notes:\t")
+		fmt.Fprintf(w, "  confinement:\t%s\n", info.Confinement)
+		if info.Broken == "" {
+			fmt.Fprintln(w, "  broken:\tfalse")
+		} else {
+			fmt.Fprintf(w, "  broken:\ttrue (%s)\n", info.Broken)
+		}
+
+	} else {
+		notes = NotesFromInfo(info)
+	}
+	fmt.Fprintf(w, "version:\t%s %s\n", info.Version, notes)
+	if info.Type != snap.TypeApp {
+		fmt.Fprintf(w, "type:\t%s\n", info.Type)
+	}
+
+	return true
+}
+
+func coalesce(snaps ...*client.Snap) *client.Snap {
+	for _, s := range snaps {
+		if s != nil {
+			return s
+		}
+	}
+	return nil
+}
+
+func (x *infoCmd) Execute([]string) error {
+	cli := Client()
+
+	w := tabwriter.NewWriter(Stdout, 2, 2, 1, ' ', 0)
+
+	noneOK := true
+	for i, snapName := range x.Positional.Snaps {
+		if i > 0 {
+			fmt.Fprintln(w, "---")
+		}
+
+		if tryDirect(w, snapName, x.Verbose) {
+			noneOK = false
+			continue
+		}
+
+		remote, _, _ := cli.FindOne(snapName)
+		local, _, _ := cli.Snap(snapName)
```

Why are errors being ignored here? Doesn't feel like a good idea.